Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -496,6 +553,16 @@ ThrowCompletionOr<Value> Value::to_number(GlobalObject& global_object) const
             return js_infinity();
         if (string == "-Infinity")
             return js_negative_infinity();
+
+        auto result = parse_number_text(string);
```

Comment:
And here you could add a reference/link/spec comments to https://tc39.es/ecma262/#sec-stringtonumber and the fact it uses a [StringNumericLiteral](https://tc39.es/ecma262/#prod-StringNumericLiteral)